### PR TITLE
#23267 Move calculate IAA queue job to IAA queue worker

### PR DESCRIPTION
**Pull request by lucassusanto**
# 💡 Root Cause <a href="#root-cause" id="root-cause" />

- Calculate IAA job queue is a long running job. It takes background queue worker resource thus blocking another jobs that are using the same queue worker

# 🌊 Changes <a href="#changes" id="changes" />

- Move calculate IAA job from background to IAA queue worker

### CODE REVIEW

File: charts/datasaur/templates/main/backend-worker-iaa-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+              value: "true"
+            - name: ENABLE_MESSAGE_QUEUE_CONSUMER
+              value: "true"
+            - name: CONSUMED_QUEUES
+              value: {{ .Values.services.worker.iaa.consume }}
+            - name: RABBITMQ_PREFETCH_COUNT_ACTION
+              value: "1"
+            - name: RABBITMQ_PREFETCH_COUNT_BACKGROUND
+              value: "1"
+            - name: RABBITMQ_PREFETCH_COUNT_CRITICAL
```
Comment (tonoman3g) [inline charts/datasaur/templates/main/backend-worker-iaa-deployment.yaml:82]: PREFETCH nya kurang yg LLM_CRITICAL @lucassusanto

**Comment by tonoman3g** [comment]
@daniel-jeremia please last check, i believe everything looks good

File: charts/datasaur/values.yaml [hunk 2]
```diff
@@ -97,6 +97,24 @@ services:
         targetCPUUtilizationPercentage: 80
         targetMemoryUtilizationPercentage: 80
       consume: external,action
+    iaa:
+      enabled: true
```
Comment (daniel-jeremia) [inline charts/datasaur/values.yaml:101]: ini emg mau on by default ya?

File: charts/datasaur/templates/main/backend-worker-iaa-deployment.yaml (hunk 1 above)
Comment (lucassusanto) [reply charts/datasaur/templates/main/backend-worker-iaa-deployment.yaml:82]: Dari kodingan sebelumnya, RABBITMQ_PREFETCH_COUNT_LLM_CRITICAL cuma ada di deployment llm-critical-worker. Apa prefetch llm criticalnya mau ditambah ke semua worker aja? Aku lebih prefer prefetch llm criticalnya ditambah ke semua worker sih

File: charts/datasaur/values.yaml (hunk 2 above)
Comment (lucassusanto) [reply charts/datasaur/values.yaml:101]: Kyknya buat self-hosted calculate IAA job masih bisa nebeng di background worker kyk action. Jadi k8s engga perlu IAA worker bahkan. Wdyt? @tonoman3g

File: charts/datasaur/templates/main/backend-worker-iaa-deployment.yaml (hunk 1 above)
Comment (tonoman3g) [reply charts/datasaur/templates/main/backend-worker-iaa-deployment.yaml:82]: ditambahin ke semua worker aja

File: charts/datasaur/values.yaml (hunk 2 above)
Comment (tonoman3g) [reply charts/datasaur/values.yaml:101]: gpp deh, on by default, 1 pod aja soale

Comment (lucassusanto) [reply charts/datasaur/values.yaml:101]: Oke

Comment (lucassusanto) [reply charts/datasaur/values.yaml:101]: Oke. On by default berarti @daniel-jeremia

File: charts/datasaur/templates/main/backend-worker-iaa-deployment.yaml (hunk 1 above)
Comment (lucassusanto) [reply charts/datasaur/templates/main/backend-worker-iaa-deployment.yaml:82]: Added

**Review by tonoman3g — APPROVED** [review]

**Comment by tonoman3g** [comment]
eh tgz nya belom diupdate lagi ya di last commit @lucassusanto

**Comment by lucassusanto** [comment]
> eh tgz nya belom diupdate lagi ya di last commit @lucassusanto

Wah iya. Updated.

**Pull request merged**